### PR TITLE
Card shuffling backend

### PR DESCRIPTION
**Pull request by hasan-umanitoba**

### CODE REVIEW

File: digital-deck-server/services/cardShuffleService.js
Comment (mdangnguyen) [inline digital-deck-server/services/cardShuffleService.js:9]: I'm not sure if it's a good idea to modify the parameter directly and return it. I'd recommend creating a new variable, modify that, and then return that as the result.

File: digital-deck-server/services/cardShuffleService.js
Comment (mdangnguyen) [inline digital-deck-server/services/cardShuffleService.js:21]: Since you're only returning 1 object, you can just do `return cards;` instead of a json so in `cardDistService` you won't have to `.card` to get the values.

File: digital-deck-server/services/cardDistService.js
Comment (mdangnguyen) [inline digital-deck-server/services/cardDistService.js:58]: There's only 1 parameter so you can just do `cardShuffleService.shuffleCards(cards)` in 1 line.

File: digital-deck-server/services/cardShuffleService.js
Comment (mdangnguyen) [inline digital-deck-server/services/cardShuffleService.js:8]: We use camel case for JS so this should be `shuffleCards`

File: digital-deck-server/services/cardShuffleService.js
Comment (hasan-umanitoba) [reply digital-deck-server/services/cardShuffleService.js]: I agree.

File: digital-deck-server/services/cardShuffleService.js
Comment (hasan-umanitoba) [reply digital-deck-server/services/cardShuffleService.js]: Done.

**Review by mdangnguyen — APPROVED** [review]
Don't merge this yet though. Pls wait until Jasdeep merge his backend PR in and rebase your branch onto dev to make sure the project structure is consistent.

File: digital-deck-server/api/controller.js
Comment (suryarawat) [inline digital-deck-server/api/controller.js:46]: Leaving a comment here as reminder. Since Dang added a router, this needs to be changed to returning 400 request. While merging if you have time please consider this.

File: digital-deck-server/services/cardShuffleService.js
Comment (suryarawat) [inline digital-deck-server/services/cardShuffleService.js:18]: I don't think having only 1 card should be a failure case. Maybe with 0 cards it makes more sense. It would be nice if you just return that 1 card as it is in this case.

File: digital-deck-server/services/cardShuffleService.js
Comment (suryarawat) [inline digital-deck-server/services/cardShuffleService.js:11]: I am curious as to why just randomly swap with only remaining values and not the whole array? ``` Math.floor(Math.random() * i); ``` . Talking about using ```i``` here. Is it better this way?

File: digital-deck-server/services/cardShuffleService.js
Comment (hasan-umanitoba) [reply digital-deck-server/services/cardShuffleService.js:18]: okay

**Pull request merged**
